src/examples/source/SmartPeakGUI.cpp

```diff
@@ -694,7 +700,77 @@ int main(int argc, char** argv)
       
       ImGui::EndMainMenuBar();
     }
+      
+    // ======================================
+    // Server-Side Execution
+    // ======================================
+      if (run_workflow_widget_->server_fields_set)
```

Comment:
I have packed some of the lines into separate functions, it's not possible to do that for the whole block since we don't want to have the gRPC library as part of the libSmartPeak or libSmartPeakWidgets.